starknet_devnet/blueprints/rpc/misc.py

```diff
@@ -27,12 +60,47 @@ async def syncing() -> Union[dict, bool]:
     return False
 
 
-# pylint: disable=redefined-builtin
-async def get_events(filter: dict) -> dict:
+# pylint: disable=too-many-arguments
+async def get_events(
+    from_block: BlockId,
+    to_block: BlockId,
+    address: Address = None,
+    keys: List[Address] = None,
+    chunk_size: int = 0,
+    continuation_token: str = "",
+) -> str:
     """
-    Returns all events matching the given filter
+    Returns all events matching the given filters.
+
+    In our implementation continuation_token is just a number.
+
+    In state.starknet_wrapper.get_state().events there is no relation between blocks.
+    This is why we need to iterate block by block, take all events,
+    and chunk it later which is not an optimal solution.
     """
-    raise NotImplementedError()
+    events = []
+    keys = [] if keys is None else [int(k, 0) for k in keys]
+    to_block = (
+        int(state.starknet_wrapper.blocks.get_number_of_blocks())
+        if to_block == "latest"
+        else int(to_block) + 1
+    )
+    for block_number in range(int(from_block), to_block):
+        block = state.starknet_wrapper.blocks.get_by_number(block_number)
+        if block.transaction_receipts != ():
+            events.extend(get_events_from_block(block, address, keys))
+
+    if chunk_size > 0 and continuation_token == "":
+        events = events[:chunk_size]
+    elif chunk_size > 0 and continuation_token != "":
+        events = (list(islice(events, chunk_size * (int(continuation_token)), None)))[
+            :chunk_size
+        ]
+
+    return RpcEventsResult(
+        events=events,
+        continuation_token=continuation_token,
```

Comment:
Should `continuation_token + 1` be returned?

Reply:
Yes as we talked about yesterday. Fixed.